src/cmds/qsub.c

```diff
@@ -6041,12 +6074,17 @@ daemon_submit(int *daemon_up, int *do_regular_submit)
 				 * We were able to send data to the background daemon.
 				 * Now, even if we fail to read back response from
 				 * background, we do not want to submit again.
+				 * If -V option is detected in background qsub, then
+				 * do a regular submit.
 				 */
-				*do_regular_submit = 0;
+				if(rc != VOPT_ERROR)
```

Comment:
Instead of checking whether VOPT_ERROR is not set, do a check for rc == VOPT_ERROR, if yes bypass the following code. Basically you need to optimise the code here.